Payjoin-spec.md

```diff
@@ -0,0 +1,61 @@
+# Payjoin in BTCPay Server
+
+## Introduction
+
+Payjoin is a generic term for any privacy mechanism where a merchant can protect his own privacy and the privacy of his customers by mixing his own UTXO to the payment of the customer.
+
+Bustapay is an specification of such mechanism. We identified some issues with the specification so we needed to deviate from it.
+
+## Difference with Bustapay
+
+The [bustapay (BIP79)](https://github.com/bitcoin/bips/blob/master/bip-0079.mediawiki) proposal has been written for payjoin.
+
+We initially implemented it, but found out that the specification was missing certain important aspect.
+Also, because this is a new proposal, we expect that even our own proposal will experience breaking changes in the following week and months.
+As such, we preferred writting our own specification for the time being.
+Once the specification stabilize and have gone through review and testing by the community, we will decide whether we will submit to the BIP process, or if we can just modify bustapay BIP79.
+
+Bustapay is currently not widely deployed, so we are not worried about breaking backward compatibility.
+
+Our implementation is identical to bustapay with the following differences.
+```diff
+- Step 1. Sender creates a bitcoin transaction paying the receiver
++ Step 1. Sender creates a PSBT paying the receiver
+
+- This transaction must use segwit for all inputs, and be fully valid and signed. The transaction must be eligible for propagation on the network (but not done so at this stage) 
++ The PSBT must be finalized.
+
+- Step 2. Sender gives the "template transaction" to the receiver
++ Step 2. Sender gives the "original PSBT" to the receiver
+
+- This is done via an HTTP POST request, sent to a "[bustapay] url" 
++ This is done via an HTTP POST request, sent to a "[payjoin] url" 
+
+Step 3. Receiver processes the transaction and returns a partially signed coinjoin
+
+- The receiver validates the transaction, and pays himself. The receiver then adds one or more of his own inputs (known as the contributed inputs) and (optionally) increases the output that pays himself (generally by the sum of the contributed inputs). Doing so creates a partial transaction, which the receiver returns to the sender. It is called such as it requires the sender to re-sign his own inputs. 
++ The receiver extracts the transaction, optionally verifying it is valid. (by calling testmempoolaccept on the extracted transaction)
++ The receiver extracts the feerate, add one of his input and decrease the change output of the payer by the fee his input added such that the feerate is the same as the "orginal PSBT".
```

Comment:
Modifying the output value on transaction submission ( excluding appending the contributed input amount) is very shady, even if just for good intentions. I really think this should be handled prior at the sender level when constructing the transaction.
Just specify a higher fee to be safe

Reply:
Why? For the sender it is easy to check that the fee rate did not changed.